web/src/components/tasks/environment-questionnaire/CheckWastePermits.test.tsx

```diff
@@ -0,0 +1,84 @@
+import { getMergedConfig } from "@/contexts/configContext";
+import { generateTask } from "@/test/factories";
+import { currentBusiness, WithStatefulUserData } from "@/test/mock/withStatefulUserData";
+import {
+  Business,
+  generateEnvironmentData,
+  generateWasteQuestionnaireData,
+  randomInt,
+} from "@businessnjgovnavigator/shared";
+import { generateBusiness, generateUserDataForBusiness } from "@businessnjgovnavigator/shared/test";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent, { UserEvent } from "@testing-library/user-event";
+import { CheckWastePermits } from "./CheckWastePermits";
+
+const Config = getMergedConfig();
+
+describe("<CheckWastePermits />", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  const task = generateTask({});
+
+  const renderQuestionnaireAndSetupUser = (business?: Business): { user: UserEvent } => {
+    render(
+      <WithStatefulUserData initialUserData={generateUserDataForBusiness(business ?? generateBusiness({}))}>
+        <CheckWastePermits task={task} />
+      </WithStatefulUserData>
+    );
+    const user = userEvent.setup();
+    return { user };
+  };
+
+  const renderQuestionnaire = (business?: Business): void => {
+    render(
+      <WithStatefulUserData initialUserData={generateUserDataForBusiness(business ?? generateBusiness({}))}>
+        <CheckWastePermits task={task} />
+      </WithStatefulUserData>
+    );
+  };
+
+  it("displays the results page when the user submits the questionnaire", async () => {
+    const { user } = renderQuestionnaireAndSetupUser();
+    await user.click(
+      screen.getByLabelText(
+        Config.wasteQuestionnaireQuestionsPage.wasteQuestionnaireOptions.hazardousMedicalWaste
+      )
+    );
+    await user.click(screen.getByText(Config.wasteQuestionnaireQuestionsPage.buttonText));
+    expect(currentBusiness().environmentData?.waste?.submitted).toBe(true);
+    await waitFor(() => {
+      expect(screen.getByText(Config.wasteQuestionnaireResultsPage.title)).toBeInTheDocument();
+    });
+  });
+
+  it("displays the questionnaire if submitted is false or undefined", async () => {
```

Comment:
If we got rid of the `undefined` case for submitted this test would no longer need to rely on randomness

Reply:
great call out!